how_to_add_new_hotspot.md

```diff
@@ -0,0 +1,142 @@
+
+#### MAC
+This parameter defines which network interface is used for MAC (Media Access Control) code. It has to be found in the devices network interfaces.
+
+#### STATUS
+This is the hotspots status LED's output GPIO index, which is usually a single green LED on most SBC's.
+
+#### BUTTON
+This parameter defines the GPIO input for Bluetooth advertisement start.
+
+Hotspots needed to be accessed via Bluetooth sometimes. Especially at the beginning, onboarding time. But keeping the required Bluetooth interface on all the time imposes two problems. First, it is a security thread and leaks some information to trace the device. Second it also effects power consumption negatively as the device is expected to work 7/24.
+
+This input triggers the Bluetooth advertisement process and then the user can access it via Bluetooth interface.
+
+#### ECCOB
+Indicates whether the ECC is onboard or not. It is a bool parameter.
+
+#### TYPE
+Defines Helium Miner type (e.g. Full, Lite etc.)
+
+#### CELLULAR
+Defines whether the device has cellular (mobile) hardware or not. It's a bool parameter.
+
+#### FCC_IDS, IC_IDS
+ID's given to the device by the regulatory bodies.
+
+#### CONTAINS_FCC_IDS, CONTAINS_IC_IDS
+The regulatory body ID's which could be found on inner modules like a cellular modem or concentrator module.
+
+## Adding automation steps
+If the new hotspot type would need a new fleet, as we do right now, it would need some changes on our automation pipeline.
+
+The developer should add necessary steps into the GitHub Action files which could be found [here](https://github.com/NebraLtd/helium-miner-software/tree/master/.github/workflows)
+
+## Modifying settings.ini
+The new hotspot could need to define another I2C bus path. Then it needed to be added into the settings.ini file and `gen_docker_compose.py` file needs to be updated, if necessary.
+
+## Defining configuration parameters in the Balena Fleet
+A single board computer generally wouldn't have a BIOS due to constraints. Therefore manufacturers like Raspberry Pi Foundation created a clever way to keep boot parameters on the device. For example there is a `config.txt` file under `/boot` for Rasspberry Pi SBC's and it defines the parameters which is cruical to make the device work properly.
+
+In Balena, such critical parameters are defined in Fleet Configuration and passed down to the device. So before adding a new device type into a new fleet configuration parameters need to be set in Fleet Configuration.
+
+## Defining variant and frequency in the Belana Fleet
+There are two critical parameters which has to be defined in a Balena Fleet before adding a new device into it.
+
+### VARIANT
+This parameter defines which variant will be used in the whole firmware stack. The name is the key from the `variant_definitions` dictionary which is detailed above.
+
+**Example:**
+| Name | Value | Service 
+|------|-------|--------
+| VARIANT | NEBHNT-IN1 | All Services
+
+### FREQ
```

Comment:
we may want to only have this for nebra fleets (or maybe not at all in future) but leave it in for now and we can always update in the future perhaps.

or maybe we could just put a note that this will probably be only for nebra fleets? not sure

Reply:
AFAIK this is required for other types, too. At least I remember so from RAK. Without this definition diagnostics container would fail to report anything. (Was planning to fix this via reporting this missing env var)